### ISSUE TITLE
Activity Timestamps

### ISSUE THREAD

**Issue by Favna** [discordjs/discord.js#2364]
**Please describe the problem you are having in as much detail as possible:**

---

Update: I found *what* causes the issue, erlpack.

Everything works just fine until you add discord's erlpack as a dependency which ends up breaking it. As for why erlpack breaks it.. honestly that's far beyond my knowledge. It may be something on Discord's side or the way it is implemented in the library I suppose.

Naturally the temporary fix is removing erlpack as it is not a must-have package anyway but that doesn't make it any less of a bug of course.

---

I seem to be getting Invalid Date for all activity timestamps no matter the user whom i request of from (playing, listening to spotify or custom set) on the latest commit of Master. I figured I'd ask here this is not just me before making a github issue on it. If someone could verify that be great.

![](https://favna.s-ul.eu/OvzdCmcI.png)

first one from previous screenshot: a spotify one but the person has since stopped listening
second one from the previous screenshot:

![](https://favna.s-ul.eu/y1Kc3DJD.png)

tried another one:

![](https://favna.s-ul.eu/nNoci4dh.png) ![](https://favna.s-ul.eu/OZr2gEI6.png)

**Include a reproducible code sample here, if possible:**
```js
const activity = message.author.presence.activity;

if (activity.timestamps) {
    return console.log(activity.timestamps);
}
return console.log("that user has no timestamps");
```

```js
// with an eval command:

eval message.guild.members.get('<some_member_id>').presence.activity.timestamps
```

**Further details:**

- discord.js version: master branch
- node.js version: 8.9.4
- Operating system: Windows 10
- Priority this issue should have – please be realistic and elaborate if possible:  Low - dependency issue

- [ ] I found this issue while running code on a __user account__
- [X] I have also tested the issue on latest master, commit hash: 35babc706d6351d5899aa6eb1ee4bf54be88cf04

**Comment by lloti** [comment]
Can reproduce

**Comment by Favna** [comment]
Actually made it an issue. As stated before not sure what happened there. That said:

TL;DR: lewdcario on discord was not able to reproduce. I'm going to do some more research. Will update as I find more info.

**Comment by Favna** [comment]
Update: I found *what* causes the issue, erlpack.

Everything works just fine until you add discord's erlpack as a dependency which ends up breaking it. As for why erlpack breaks it.. honestly that's far beyond my knowledge. It may be something on Discord's side or the way it is implemented in the library I suppose.

Naturally the temporary fix is removing erlpack as it is not a must-have package anyway but that doesn't make it any less of a bug of course.

**Comment by bdistin** [comment]
reproduced bug by installing erlpack: ![img](https://puu.sh/zxaaR/d47559bdf7.png)
